### PR TITLE
integration: Fix cleaning up resources when tests complete

### PR DESCRIPTION
**Pull request by mauriciovasquezbernal**
Yet another iteration on this one. The goal of TestMain() is to setup
Inspektor Gadget before running the test and tear it down after.
d061e9e4cd74 ("integration: Fix cleaning up commands") switched to use
defer in order to perform the clean up when the test fails. (It ends up
calling runtime.Goexit() which executes all defer calls). It worked fine
for that case but then when the test succeded it wasn't working fine
because os.Exit() doesn't call the defered functions.

This commit switches to an approach that mixes both of them, defer is
used in case the test fails and the cleanup functions are explictly
called before os.Exit().

### CODE REVIEW

File: integration/integration_test.go [hunk 1]
```diff
@@ -87,26 +87,31 @@ func TestMain(m *testing.M) {
...
-	// fails (hence calling runtime.Goexit())
-	defer func() {
-		fmt.Printf("Clean inspektor-gadget:\n")
-		cleanupTestNamespace.runWithoutTest()
-		cleanupInspektorGadget.runWithoutTest()
-	}()
-
 	ret := m.Run()
 
+	// os.Exit() doesn't call defered functions, hence do the cleanup manually.
```
Comment (eiffel-fl) [inline integration/integration_test.go:112]: If the problem comes from `os.Exit()`, what about replacing it by something like this:

```golang
if ret != 0 {
  t.Fatal("return code is %v", ret)
}
```

**Review by eiffel-fl — APPROVED** [review]
Thank you for it!
These integration tests are more complicated than first estimation...

I am OK for it to be merged like this but can you please take a look at my comment before?

File: integration/integration_test.go (hunk 1 above)
Comment (mauriciovasquezbernal) [reply integration/integration_test.go:112]: I don't think that's going to work. `os.Exit()` sets the return code of the application and I'm not sure `t.Fatal` does the same. Also the `TestMain` documentation recommends to use `m.Run` with `os.Exit()`.

Comment (eiffel-fl) [reply integration/integration_test.go:112]: `t.Fatal()` would make the application returning something different than 0, I think.
But you are right, the document advices to do something like this `os.Exit(m.Run())`.

You can merge whenever you want, I already rebased my PR on top of yours.

**Pull request merged**